src/services/expenseService.js

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable function-paren-newline */
+const { createUniqueID } = require('../utils/createUniqueID');
+
+let expenses = [];
+
+const initExpenses = () => {
+  expenses = [];
+};
+
+const getAllExpenses = (id, categories, from, to) => {
+  let preparedExpenses = expenses;
+
+  if (id) {
+    preparedExpenses = preparedExpenses.filter(({ userId }) => userId === id);
+  }
+
+  if (categories) {
+    preparedExpenses = preparedExpenses.filter(({ category }) =>
+      categories.includes(category),
+    );
+  }
```

Comment:
You use filter at least three times here. is it possible to combine it to iterate only once?